packages/rocketchat-ui-flextab/client/tabs/messageSearch.js

```diff
@@ -8,8 +8,16 @@ Meteor.startup(function() {
 		],
 		action() {
 			const message = this._arguments[1];
+			if (Session.get('openedRoom') === message.rid) {
+				return RoomHistoryManager.getSurroundingMessages(message, 50);
+			}
+			FlowRouter.goToRoomById(message.rid);
 			RocketChat.MessageAction.hideDropDown();
-			return RoomHistoryManager.getSurroundingMessages(message, 50);
+			window.setTimeout(() => {
```

Comment:
ahh, if the connection isn't very good, it sometimes takes some time to go to other room, and it was giving errors, is there a promise or callback which I can use?

I tried without `setTimeout` because this feels very dirty, it worked the first time because it was instant, then it stopped working.